### PR TITLE
feat(core): basic types

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* kotlin core SDK: basic types

## Linear Issue Reference

* LOG-633

## Testing

None.

### CODE REVIEW

File: kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+dependencies {
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.5.31"))
+    implementation(platform("com.squareup.okhttp3:okhttp-bom:4.9.3"))
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    implementation("org.bitbucket.b_c:jose4j:0.7.9")
+    implementation("com.aventrix.jnanoid:jnanoid:2.0.0")
+    implementation("com.squareup.okhttp3:okhttp")
+    implementation("com.google.code.gson:gson:2.8.9")
+}
```
Comment (gao-sun) [inline kotlin/build.gradle.kts:26]: trailing new line?

File: kotlin/src/main/kotlin/io/logto/sdk/core/type/TokenResponse.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package io.logto.sdk.core.type
+
+data class TokenResponse(
+    val accessToken: String,
+    val refreshToken: String?,
```
Comment (gao-sun) [inline kotlin/src/main/kotlin/io/logto/sdk/core/type/TokenResponse.kt:5]: This is required?

Comment (gao-sun) [reply kotlin/src/main/kotlin/io/logto/sdk/core/type/TokenResponse.kt:5]: as i recall `openid` and `offline_access` are both required in v1? @wangsijie @simeng-li

Comment (simeng-li) [reply kotlin/src/main/kotlin/io/logto/sdk/core/type/TokenResponse.kt:5]: yep

Comment (xiaoyijun) [reply kotlin/src/main/kotlin/io/logto/sdk/core/type/TokenResponse.kt]: updated

File: kotlin/src/main/kotlin/io/logto/sdk/core/type/IdTokenClaims.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.logto.sdk.core.type
+
+data class IdTokenClaims(
+    val sub: String,
+    val atHash: String,
```
Comment (gao-sun) [inline kotlin/src/main/kotlin/io/logto/sdk/core/type/IdTokenClaims.kt:5]: maybe optional according to RFC and slack discussion. pending finalization on Slack.

**Review by gao-sun — APPROVED** [review]
lgtm

File: kotlin/src/main/kotlin/io/logto/sdk/core/type/IdTokenClaims.kt (hunk 3 above)
Comment (xiaoyijun) [reply kotlin/src/main/kotlin/io/logto/sdk/core/type/IdTokenClaims.kt]: updated

**Pull request merged**